web/src/connectors/task-annotator-connector/utils.ts

```diff
@@ -0,0 +1,129 @@
+import { CategoryDataAttrType } from '../../api/typings';
+import { PageSize } from '../../shared/components/document-pages/document-pages';
+
+interface Token {
+    id: number;
+    text: string;
+    x: number;
+    y: number;
+    width: number;
+    height: number;
+}
+
+interface Data {
+    tokens: Token[];
+    dataAttributes: CategoryDataAttrType[];
+}
+
+interface PageInfoObj {
+    id: number;
+    type: string;
+    bbox: number[];
+    category: string;
+    data: Data;
+    children: number[];
+    text: string;
+}
+
+interface Page {
+    page_num: number;
+    size: PageSize;
+    objs: PageInfoObj[];
+    revision: string;
+    user_id: string;
+    pipeline: number;
+    date: string;
+    is_validated: boolean;
+    categories: string[];
+}
+
+export interface LatestRevisionObj {
+    [page_num: string]: Page[];
+}
+
+export interface RevisionObjByUser {
+    [user_id: string]: {
+        [page_num: number]: {
+            size: PageSize;
+            objs: PageInfoObj[];
+        };
+    };
+}
+
+export interface UserRevision {
+    user_id: string;
+    page_num: number;
+    size: PageSize;
+    objs: PageInfoObj[];
+    revision: string;
+    isValidated: boolean;
+    date: string;
+    pipeline: number;
+    categories: string[];
+}
+
+export function convertToRevisionByUserArray(sourceObj: LatestRevisionObj): UserRevision[] {
+    const userRevisions: UserRevision[] = [];
+
+    for (const page_num in sourceObj) {
+        const pages = sourceObj[page_num];
+
+        for (const page of pages) {
+            const {
+                user_id: user_id,
+                size,
+                objs,
+                revision,
+                is_validated,
+                date,
+                pipeline,
+                categories
+            } = page;
+
+            userRevisions.push({
+                user_id,
+                page_num: parseInt(page_num),
+                size,
+                objs,
+                revision,
+                isValidated: is_validated,
+                date,
+                pipeline,
+                categories
+            });
+        }
+    }
+
+    return userRevisions;
+}
+
+export function delAllRevisionByUser(
+    sourceObj: RevisionObjByUser,
+    user_id?: string
+): RevisionObjByUser {
+    if (!user_id) return sourceObj;
+
+    const result: RevisionObjByUser = {};
+
+    for (const key in sourceObj) {
```

Comment:
I believe we can do something like this

```
const result = {...source};
delete result[user_id]
```

Reply:
fixed https://github.com/epam/badgerdoc/pull/584/commits/5fc548e91e1f525470ad4a988b68acafeff71e18